lib/src/management_enforcer.dart

```diff
@@ -0,0 +1,432 @@
+  }
+
+  /// getAllNamedObjects gets the list of objects that show up in the current named policy.
+  ///
+  /// [ptype] the policy type, can be "p", "p2", "p3", ..
+  /// return all the objects in policy rules of the ptype type. It actually
+  /// collects the 1-index elements of the policy rules. So make sure
+  /// your object is the 1-index element, like (sub, obj, act).
+  /// Duplicates are removed.
+
+  List<String> getAllNamedObjects(String ptype) {
+    return model.getValuesForFieldInPolicy('p', ptype, 1);
+  }
+
+  /// getAllActions gets the list of actions that show up in the current policy.
+  ///
+  /// @return all the actions in "p" policy rules. It actually collects
+  ///         the 2-index elements of "p" policy rules. So make sure your action
+  ///         is the 2-index element, like (sub, obj, act).
+  ///         Duplicates are removed.
+
+  List<String> getAllActions() {
+    return getAllNamedActions('p');
+  }
+
+  /// GetAllNamedActions gets the list of actions that show up in the current named policy.
+  ///
+  /// [ptype] the policy type, can be "p", "p2", "p3", ..
+  /// return all the actions in policy rules of the ptype type. It actually
+  /// collects the 2-index elements of the policy rules. So make sure
+  /// your action is the 2-index element, like (sub, obj, act).
+  /// Duplicates are removed.
+
+  List<String> getAllNamedActions(String ptype) {
+    return model.getValuesForFieldInPolicy('p', ptype, 2);
+  }
+
+  /// getAllRoles gets the list of roles that show up in the current policy.
+  ///
+  /// return all the roles in "g" policy rules. It actually collects
+  /// the 1-index elements of "g" policy rules. So make sure your
+  /// role is the 1-index element, like (sub, role).
+  /// Duplicates are removed.
+
+  List<String> getAllRoles() {
+    return getAllNamedRoles('g');
+  }
+
+  /// getAllNamedRoles gets the list of roles that show up in the current named policy.
+  ///
+  /// [param] ptype the policy type, can be "g", "g2", "g3", ..
+  /// return all the subjects in policy rules of the ptype type. It actually
+  /// collects the 0-index elements of the policy rules. So make
+  /// sure your subject is the 0-index element, like (sub, obj, act).
+  /// Duplicates are removed.
+
+  List<String> getAllNamedRoles(String ptype) {
+    return model.getValuesForFieldInPolicy('g', ptype, 1);
+  }
+
+  /// getPolicy gets all the authorization rules in the policy.
+  ///
+  /// return all the "p" policy rules.
+
+  List<List<String>> getPolicy() {
+    return getNamedPolicy('p');
+  }
+
+  /// getFilteredPolicy gets all the authorization rules in the policy, field filters can be specified.
+  ///
+  /// [fieldIndex] the policy rule's start index to be matched.
+  /// [fieldValues] the field values to be matched, value ""
+  /// means not to match this field.
+  /// return the filtered "p" policy rules.
+
+  List<List<String>> getFilteredPolicy(int fieldIndex, dynamic fieldValues) {
```

Comment:
Declarative data types would be better for variable arguments. 
`List<String> fieldValues` or `List<dynamic> fieldValues`